src/controllers/serviceaccounts.go

```diff
@@ -154,6 +161,12 @@ func (c *serviceAccountController) DeleteServiceAccount(name, namespace string)
 		return customerrors.NewAPIError(fmt.Sprintf(ErrCouldNotDeleteServiceAccount, name, namespace), err)
 	}
 
+	err = c.client.CoreV1().Secrets(namespace).Delete(c.ctx, fmt.Sprintf("%s-%s", name, utils.TokenRequestSuffix), metav1.DeleteOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return customerrors.NewAPIError(fmt.Sprintf(ErrCouldNotDeleteTokenRequestSecret, name, namespace), err)
+		}
+	}
```

Comment:
Instead of doing an explicit delete here, maybe use an `ownerReference` on the `secret`, so that when the `serviceAccount` is deleted then Kubernetes takes care of deleting the `secret` on its own?